src/disk_io_thread.cpp

```diff
@@ -241,9 +241,9 @@ namespace libtorrent
 				// the magic number 3 is also used in add_job()
 				// every 4:th thread is a hasher thread
 				if ((thread_id & 0x3) == 3) type = hasher_thread;
-				m_threads.push_back(boost::shared_ptr<thread>(
+				m_threads.emplace_back(
```

Comment:
I believe these thread objects are no longer heap-allocated either (in the "threads" branch)